tgui/packages/tgui/interfaces/ClothingBooth/GroupingTag.tsx

```diff
@@ -0,0 +1,55 @@
+import { Box, Stack } from '../../components';
+import { buildFieldComparator, numberComparator } from './utils/comparator';
+import { ClothingBoothData, ClothingBoothGroupingTagsData, ClothingBoothSlotKey } from './type';
+import { useBackend } from '../../backend';
+
+interface GroupingTagContainerProps {
+  slot: ClothingBoothSlotKey;
+  grouping_tags: string[];
+}
+
+interface ClothingBoothSlotDetail {
+  id: ClothingBoothSlotKey;
+  name: string;
+}
+
+export const GroupingTagContainer = (props: GroupingTagContainerProps, context) => {
+  const { data } = useBackend<ClothingBoothData>(context);
+  const { tags } = data;
+  const { slot, grouping_tags } = props;
+  const groupingTagsObject = Object.values(grouping_tags);
+  const sortedGroupingTags = groupingTagsObject.sort(
+    buildFieldComparator((groupingTag) => tags[groupingTag].display_order, numberComparator)
+  );
+  const clothingBoothSlotLookup = Object.entries(ClothingBoothSlotKey).reduce((acc, [name, id]) => {
+    acc[id] = { id, name };
+    return acc;
+  }, {} as Record<ClothingBoothSlotKey, ClothingBoothSlotDetail>);
```

Comment:
This can be moved outside of the component, as it's solely based on `ClothingBoothSlotKey` which doesn't change, so it's only generated once.